docs/calling-downstream-apis/azure-sdks.md

```diff
@@ -0,0 +1,757 @@
+        
+        var secretClient = new SecretClient(
+            new Uri("https://myvault.vault.azure.net"),
+            _credential);
+        
+        var secret = await secretClient.GetSecretAsync(secretName);
+        return secret.Value.Value;
+    }
+}
+```
+
+See [Agent Identities documentation](../scenarios/agent-identities/README.md) for more details.
+
+## Managed Identity Integration
+
+`MicrosoftIdentityTokenCredential` works seamlessly with Azure Managed Identity:
+
+### Configuration for Managed Identity
+
+```json
+{
+  "AzureAd": {
+    "Instance": "https://login.microsoftonline.com/",
+    "TenantId": "your-tenant-id",
+    "ClientId": "your-client-id",
+    "ClientCredentials": [
+      {
+        "SourceType": "SignedAssertionFromManagedIdentity"
+      }
+    ]
+  }
+}
+```
+
+### Using System-Assigned Managed Identity
+
+```csharp
+// No additional code needed!
+// When deployed to Azure, the credential automatically uses managed identity
+
+public class StorageService
+{
+    private readonly MicrosoftIdentityTokenCredential _credential;
+    
+    public StorageService(MicrosoftIdentityTokenCredential credential)
+    {
+        _credential = credential;
+        _credential.Options.RequestAppToken = true;
+    }
+    
+    public async Task<List<string>> ListContainersAsync()
+    {
+        // Uses managed identity when running in Azure
+        var blobClient = new BlobServiceClient(
+            new Uri("https://myaccount.blob.core.windows.net"),
+            _credential);
+        
+        var containers = new List<string>();
+        await foreach (var container in blobClient.GetBlobContainersAsync())
+        {
+            containers.Add(container.Name);
+        }
+        
+        return containers;
+    }
+}
+```
+
+### Using User-Assigned Managed Identity
+
+```json
+{
+  "AzureAd": {
+    "ClientCredentials": [
+      {
+        "SourceType": "SignedAssertionFromManagedIdentity",
+        "ManagedIdentityClientId": "user-assigned-identity-client-id"
+      }
+    ]
+  }
+}
+```
+
+## OWIN Implementation
+
+For ASP.NET applications using OWIN:
+
+```csharp
+using Microsoft.Identity.Web;
+using Microsoft.Identity.Web.OWIN;
+using Owin;
+
+public class Startup
+{
+    public void Configuration(IAppBuilder app)
+    {
+        app.AddMicrosoftIdentityWebApp(
+            updateOptions: options =>
+            {
+                Configuration.Bind("AzureAd", options);
+            },
+            configureMicrosoftIdentityOptions: options =>
+            {
+                options.EnableTokenAcquisitionToCallDownstreamApi();
+                options.AddMicrosoftIdentityAzureTokenCredential();
+                options.AddInMemoryTokenCaches();
+            });
+    }
+}
+```
+
+## Best Practices
+
+### 1. Reuse Azure SDK Clients
+
+Azure SDK clients are thread-safe and should be reused:
+
+```csharp
+// ✅ Good: Register as singleton
+builder.Services.AddSingleton<BlobServiceClient>(sp =>
```

Comment:
This doesn't follow the guidance from Azure SDK on DI : https://learn.microsoft.com/en-us/dotnet/azure/sdk/dependency-injection?tabs=web-app-builder

This should look like:

```cs
builder.Services.AddAzureClients(clientBuilder =>
{
    // Register clients for each service
    clientBuilder.AddBlobServiceClient(new Uri("https://myaccount.blob.core.windows.net"));

    // Set a credential for all clients to use by default
    clientBuilder.UseCredential(sp => sp.GetRequiredService<MicrosoftIdentityTokenCredential>());
});
```

Reply:
I agree. thanks @keegan-caruso